### PR TITLE
Don't getCiBuildId from GitHub when it's already overridden in the action parameters

### PR DESCRIPTION
**Pull request by jdborneman-terminus**
We're doing Strange Things™️ with our setup. Long story short we have an infra repo that deploys and tests a service repo. The service repo is the one that _contains the tests_. Instead of having the infra repo call out to a workflow in the service repo (causing all sorts of interdependency between workflows) we've instead wrapped the cypress-io/github-action in a custom composite action that gets run from teh Infra repo and pulls down the service repo, testing it as part of teh infra workflow.

**_BUT_**... it fails in the following way:
```
2022-05-18T18:24:10.078Z @cypress/github-action no need to verify Cypress binary or save caches
2022-05-18T18:24:10.078Z @cypress/github-action No start command found
2022-05-18T18:24:10.079Z @cypress/github-action Running Cypress tests using CLI command
2022-05-18T18:24:10.079Z @cypress/github-action with concatenated command prefix: yarn
2022-05-18T18:24:10.079Z @cypress/github-action Determining build id by asking GitHub about run 2347025404
2022-05-18T18:24:10.159Z @cypress/github-action Not Found
2022-05-18T18:24:10.166Z @cypress/github-action HttpError: Not Found
Error: Not Found
    at /home/runner/work/_actions/cypress-io/github-action/v2/dist/index.js:58740:21
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async getCiBuildId (/home/runner/work/_actions/cypress-io/github-action/v2/dist/index.js:6230:18)
    at async runTestsUsingCommandLine (/home/runner/work/_actions/cypress-io/github-action/v2/dist/index.js:6347:33)
```

I was able to track down that it's failing because it's trying to determine a build id from the repo but I believe it's confused about which repo and run id to use since we're inside a repo-in-a-repo.

This change just make sit so that `getCiBuildId` isn't called if the `ci-build-id` parameter is already provided.

_**Note:**_ Not sure why the package-lock.json got changed so drastically. All I did was an `npm install`.

Anyway open to feedback and changes to what I did as js is not my best language (yet 😁 ).

### CODE REVIEW

**Comment by jdborneman-terminus** [comment]
https://github.com/jdborneman-terminus/cypress-io-github-action/pull/1

A PR I did on my fork showed 51 of 52 checks passed. Not sure why the 1 failed TBH:
https://github.com/jdborneman-terminus/cypress-io-github-action/runs/6495534617?check_suite_focus=true

File: index.js [hunk 1]
```diff
@@ -551,13 +551,17 @@ const runTestsUsingCommandLine = async () => {
   }
 
   if (parallel || group) {
-    const { branch, buildId } = await getCiBuildId()
-    if (branch) {
-      core.exportVariable('GH_BRANCH', branch)
+    var buildIdVar = null
```
Comment (admah) [inline index.js:554]:
```suggestion
    let buildIdVar = null
```


Use `let` here to match the rest of the code.

File: package-lock.json [hunk 2]
```diff
@@ -1,2135 +1,8 @@
 {
```
Comment (admah) [inline package-lock.json:1]: I think you need to restore this file to what it is on `master` and rerun. This appears to remove all `packages` from the lockfile.

**Review by admah — CHANGES_REQUESTED** [review]
See a couple of comments. Thanks!

File: package-lock.json (hunk 2 above)
Comment (jdborneman-terminus) [reply package-lock.json:1]: Strange it did that. Can do!

**Comment by jdborneman-terminus** [comment]
@admah So .... strange things. I reverted package-lock.json, but now when I do an npm run build it shows a LOT more changes to dist/index.js than I expected. I'm currently testing it but if you've got any advice as to WHY this might be happening I'm all ears. I'm on `npm` version `6.14.15`

**Comment by jdborneman-terminus** [comment]
@admah I upgraded npm to 8* and it's still makign pretty large changes to dist/index.js when I `npm run build` (although it now doesn't monkey with the package-lock when I npm install at least).

**Comment by admah** [comment]
@jdborneman-terminus I'm taking a look now to see what the diff is.

**Review by admah — APPROVED** [review]

**Comment by jdborneman-terminus** [comment]
Thanks @admah! Happy to be able to contribute. So when will it be tagged/released? Is that something I can/should do, or leave it to you all?

**Comment by admah** [comment]
@jdborneman-terminus I'm working on this. We follow semantic-release commit conventions and I missed that no commits seem to have `feat:`. That means no release was kicked off. I'll update this once it's updated. Thanks!

**Pull request merged**